tests/models/hf/test_fsdp_weight_tying.py

```diff
@@ -0,0 +1,97 @@
+# Copyright 2024 MosaicML LLM Foundry authors
+# SPDX-License-Identifier: Apache-2.0
+
+import pathlib
+from typing import Optional
+
+import pytest
+from composer import Trainer
+from omegaconf import OmegaConf as om
+
+from llmfoundry import COMPOSER_MODEL_REGISTRY
+from llmfoundry.utils.builders import build_tokenizer
+
+
+@pytest.mark.world_size(2)
+@pytest.mark.gpu
+@pytest.mark.parametrize('peft_config', [
+    None, {
+        'peft_type': 'LORA',
+        'task_type': 'CAUSAL_LM',
+        'lora_alpha': 32,
+        'lora_dropout': 0.05,
+        'r': 16,
+        'target_modules': [
+            'q_proj',
+            'k_proj',
+            'v_proj',
+        ],
+    }
+])
+@pytest.mark.parametrize('init_device', ['cpu', 'mixed', 'meta'])
+def test_fsdp_weight_tying(peft_config: Optional[dict], tmp_path: pathlib.Path,
+                           init_device: str):
+    model_cfg = {
+        'name': 'hf_causal_lm',
+        'pretrained_model_name_or_path': 'mistralai/Mistral-7B-v0.1',
+        'config_overrides': {
+            'num_hidden_layers': 2,
+            'hidden_size': 32,
+            'intermediate_size': 64,
+            'tie_word_embeddings': True,
+        },
+        'use_auth_token': True,
```

Comment:
does it need to be True?
```suggestion
```

Reply:
if so, do we have a HF auth token for ci cd?

Reply:
ah, no, just copy paste, will remove.

Reply:
there is a jira somewhere to get ci to have hf token so we can test llama in ci.